src/java.base/share/classes/java/lang/ref/Reference.java

```diff
@@ -354,9 +354,16 @@ public void runFinalization() {
      */
     @IntrinsicCandidate
     public T get() {
-        return this.referent;
+        @SuppressWarnings("unchecked")
+        T result = (T) get0();
+        return result;
     }
 
+    /* Implementation of unintrinsified get().  Making get() native may lead
+     * C2 to sometimes prefer the native implementation over the intrinsic.
+     */
+    private native Object get0();
```

Comment:
I think you can declare this as `private native T get0();` without changes to native method signatures, so you can avoid the unchecked cast above. (See Class::getPrimitiveClass declaration)

Also, can C2 choose to use native over intrinsic? That is concerning from a performance POV, as I think there are a few such performance sensitive methods in core libraries.

Reply:
It hadn't occurred to me that would work, but indeed it does.  Thanks for the suggestion.

I've used tools like `-XX:+PrintInlining` when running a test that I found to be performance sensitive to
the use of the intrinsic, and found that it was used.